src/dmtcp_dlsym.cpp

```diff
@@ -282,33 +280,59 @@ get_dt_tags(void *handle, dt_tag *tags)
 
   ElfW(Dyn) * cur_dyn;
 
+/*
+ * For libraries loaded by RTLD, the DT_* entries are patched at load time.
+ * Here's an example call trace:
+ *   _start()
+ *     ...
+ *     dl_main()
+ *       ...
+ *       _dl_map_object()
+ *         _dl_map_object_from_fd()
+ *           _dl_map_segments()
+ *           elf_get_dynamic_info()
+ *
```

Comment:
Let's add a small comment:  This code extends dmtcp_dlsym to work on VDSO, while in libc, they have a separate internal vdso_dlsym function for this purpose.